### ISSUE TITLE
[a11y]: ToggletipButton automatically injects aria-describedby, causing screen reader verbosity for complex content

### ISSUE THREAD

**Issue by luizas** [carbon-design-system/carbon#22445]
### Package

@carbon/react

### Browser

Chrome, Edge

### Operating System

MacOS, Windows

### Package version

carbon/react:1.99.0

### React version

v19.2.3

### Automated testing tool and ruleset

IBM Accessibility Checker

### Assistive technology

NVDA

### Description

According to the W3C ARIA Authoring Practices, a Toggletip containing complex content should follow the Disclosure pattern, relying on aria-expanded and aria-controls. It should not use aria-describedby.
Because Carbon forces aria-describedby by default, screen readers (specifically NVDA) read the entire content of the popover the moment the trigger button receives keyboard focus, before the user has even opened it. When placed inside complex components like a DataTable, this creates an overwhelming and confusing experience, often resulting in the content being announced multiple times.

### WCAG 2.1 Violation

4.1.2 and 1.3.1

### Reproduction/example

https://stackblitz.com/edit/vitejs-vite-pjfi11dg?file=src%2FApp.jsx,src%2Findex.css,src%2FApp.css,index.html&terminal=dev

### Steps to reproduce

Steps to Reproduce

- Create a Toggletip component containing complex content
- Place this Toggletip inside a DataTable component
- Turn on a screen reader (NVDA)
- Navigate to the ToggletipButton using the keyboard (Tab key)

Actual Behavior

- The screen reader immediately reads the entire complex content of the popover three times

Expected Behavior

- The screen reader should read the popover content once

<img width="655" height="547" alt="Image" src="https://github.com/user-attachments/assets/f163f376-13fa-46e7-9383-14b861332f15" />
<img width="901" height="713" alt="Image" src="https://github.com/user-attachments/assets/1369b539-f0e7-44c4-addd-2ec98df278b9" />

### Suggested Severity

None

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
